src/solvers/anderson.jl

```diff
@@ -14,34 +12,40 @@ struct AndersonCache{Tx,To,Tdg,Tg,TQ,TR} <: AbstractSolverCache
     R::TR
 end
 
-function AndersonCache(df, ::Anderson{m}) where m
+function AndersonCache(df, m)
     x = similar(df.x_f)
     g = similar(x)
 
-    fxold = similar(x)
-    gold = similar(x)
-
-    # maximum size of history
-    mmax = min(length(x), m)
-
-    # buffer storing the differences between g of the iterates, from oldest to newest
-    Δgs = [similar(x) for _ in 1:mmax]
-
-    T = eltype(x)
-    γs = Vector{T}(undef, mmax) # coefficients obtained from the least-squares problem
-
-    # matrices for QR decomposition
-    Q = Matrix{T}(undef, length(x), mmax)
-    R = Matrix{T}(undef, mmax, mmax)
+    if m > 0
+        fxold = similar(x)
+        gold = similar(x)
+
+        # maximum size of history
+        mmax = min(length(x), m)
+
+        # buffer storing the differences between g of the iterates, from oldest to newest
+        Δgs = [similar(x) for _ in 1:mmax]
+
+        T = eltype(x)
+        γs = Vector{T}(undef, mmax) # coefficients obtained from the least-squares problem
+
+        # matrices for QR decomposition
+        Q = Matrix{T}(undef, length(x), mmax)
+        R = Matrix{T}(undef, mmax, mmax)
+    else
+        fxold = nothing
+        gold = nothing
+        Δgs = nothing
+        γs = nothing
+        Q = nothing
+        R = nothing
+    end
```

Comment:
This is unstable. Not sure if that's a big problem in practice since small type unions are supposed to be fast these days, and anyway Anderson is mostly useful for large scale in which the overhead doesn't matter...